library/mr_provisioner_machine_provision.py

```diff
@@ -115,8 +115,11 @@ def set_machine_parameters(url, token, machine_id, initrd_id=None,
         parameters['preseed_id'] = preseed_id
     if subarch:
         parameters['subarch'] = subarch
+    if kernel_opts:
+        parameters['kernel_opts'] = kernel_opts
+    else:
```

Comment:
Why is this else necessary when none of the other parameters use it? Please be consistent with the rest of the options, unless this case is in fact different.